websites/M/MangaHere/presence.ts

```diff
@@ -0,0 +1,144 @@
+const presence = new Presence({
+  clientId: "831262912815300638"
+}),
+  browsingStamp = Math.floor(Date.now() / 1000);
+
+// let username: string;
+
+presence.on("UpdateData", async () => {
+  const data: PresenceData = {
+    largeImageKey: "mangahere"
+  },
+    pathname = document.location.pathname,
+    hostname = document.location.hostname,
+    ganres = ["martial-arts", "action", "school-life",
+              "sci-fi", "yoi", "shotacon", "mystery", "shoujo",
+              "ecchi", "doujinshi", "lolicon", "adventure", "romance",
+              "gender-bender", "harem", "sports", "webtoons", "comedy",
+              "shounen-ai", "josei", "shoujo-ai", "adult", "fantasy", "supernatural",
+              "psychological", "yuri", "one-shot", "historical", "drama",
+              "seinen", "mature", "smut", "horror", "shounen", "slice-of-life",
+              "tragedy", "mecha"
+            ];
+
+  if (hostname === "mangahere.cc" || hostname === "www.mangahere.cc") {
+
+    if (pathname == "/") {
+      data.details = "Viewing the Homepage";
+      data.startTimestamp = browsingStamp;
+      data.buttons = [{ label: "Visit the website", url: "www.mangahere.cc" }];
+    }
+    else if (pathname == "/latest/") {
+      data.details = "Browsing latest manga";
+      data.startTimestamp = browsingStamp;
+    }
+    else if (pathname == "/ranking/") {
+      data.details = "Browsing by ranking";
+      data.startTimestamp = browsingStamp;
+    }
+    else if (pathname == "/spoilers/") {
+      data.details = "Browsing spoilers and news";
+      data.startTimestamp = browsingStamp;
+    }
+    else if (pathname == "/directory/") {
+      data.details = "Browsing all manga";
+      data.startTimestamp = browsingStamp;
+    }
+    else if (pathname == "/on_going/") {
+      data.details = "Browsing ongoing manga";
+      data.startTimestamp = browsingStamp;
+    }
+    //ganre/new/
+    else if (pathname.endsWith("/new/")) {
+      const splitUrl = url.split('/');
+      if (splitUrl[1] == "new") {
+        data.details = "Browsing new manga";
+      } else {
+        data.details = "Browsing new " + splitUrl[1] + " manga";
+      }
+      data.startTimestamp = browsingStamp;
+    }
+    //ganre/completed/
+    else if (pathname.endsWith("/completed/")) {
+      let splitUrl = url.split('/');
```

Comment:
Your consistency of using const or let is bad, why don't you just stick to const as the value is never reassigned?

Reply:
Yeah i just became aware of it, I will stick to const.